app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessTradeNameController.scala

```diff
@@ -0,0 +1,86 @@
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request}
+import play.twirl.api.Html
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.IncomeTaxSubscriptionConnector
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessTradeNameForm._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.FormUtil._
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.BusinessTradeNameModel
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.services.AuthService
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.utilities.ImplicitDateFormatter
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.views.html.business_trade_name
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.play.language.LanguageUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+@Singleton
+class BusinessTradeNameController @Inject()(mcc: MessagesControllerComponents,
+                                            incomeTaxSubscriptionConnector: IncomeTaxSubscriptionConnector,
+                                            authService: AuthService,
+                                            val languageUtils: LanguageUtils)
+                                           (implicit val ec: ExecutionContext, val appConfig: AppConfig) extends FrontendController(mcc)
+  with I18nSupport with ImplicitDateFormatter {
+
+  def view(businessTradeNameForm: Form[BusinessTradeNameModel])(implicit request: Request[AnyContent]): Html =
+    business_trade_name(
+      businessTradeNameForm = businessTradeNameForm,
+      postAction = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessTradeNameController.submit(),
+      backUrl = backUrl()
+    )
+
+
+  def show(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      incomeTaxSubscriptionConnector.getSelfEmployments[BusinessTradeNameModel](BusinessTradeNameController.businessTradeNameKey).map {
+        case Right(businessTradeName) =>
+          Ok(view(businessTradeNameValidationForm.fill(businessTradeName)))
+        case error => throw new InternalServerException(error.toString)
+      }
+    }
+  }
+
+
+  def submit(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      businessTradeNameValidationForm.bindFromRequest.fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors))),
+        businessTradeName =>
+          incomeTaxSubscriptionConnector.saveSelfEmployments(BusinessTradeNameController.businessTradeNameKey, businessTradeName) map (_ =>
+            Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessStartDateController.show())
```

Comment:
```suggestion
            Redirect(uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessTradeNameController.show())
```
Can this redirect to it's own show method?